### DISCUSSION TITLE
[F] JS float precision

### DISCUSSION THREAD

**Discussion by VeblenDian** [exceljs/exceljs#2454] [Ideas & Proposals]
## 🚀 Feature Proposal

When some decimals are entered in Excel, the numbers parsed by Exceljs will lose precision, such as 0.15 -> 0.1500000000002Ï

## Motivation

Please outline the motivation for the proposal.

## Example

Please provide an example for how this feature would be used.

**Comment by zurmokeeper** [comment]
@VeblenDian Can you provide a specific example? Or provide an excel template?

**Comment by krismorf** [comment]
I believe this is happening because of [JavaScript number's representation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Numbers_and_dates#:~:text=In%20JavaScript%2C%20numbers%20are%20implemented,1%20can%20be%20represented%20exactly.).

![exceljs-float](https://github.com/exceljs/exceljs/assets/42035849/abcfe3c3-62de-4f14-99d4-5d93d5d33e20)

The `xlsx` package has these options available `{ rawNumbers: false, raw: true }` which handle this problem.

**Comment by chrishearn** [comment]
I've just run into this same problem and had to round the numbers manually in my application code to get them to match the original Excel cell value. Would be great to get this resolved in this package as this is quite odd and unexpected behaviour.

**Comment by lulunac27a** [comment]
The numbers cannot be expressed in exact decimal form because JavaScript uses 64-bit double precision floating point numbers, like 0.15 which cannot be expressed in exact binary form, so it will round to nearest 53-bit significand precision.

**Reply by deepakprakash007** [reply]
Is there no other way I can get the exact value/text what we see in the excel sheet?

**Reply by lulunac27a** [reply]
Use the `TEXT()` function in Excel to see the exact numeric value as text, with full significant digits/decimal places precision required.
